kartothek/io_components/read.py

```diff
@@ -65,8 +73,11 @@ def dispatch_metapartitions_from_factory(
         assert base_df is not None
         base_df = cast(pd.DataFrame, base_df)
 
-        # Group the resulting MetaParitions by partition keys
-        merged_partitions = base_df.groupby(dataset_factory.partition_keys)
+        # Group the resulting MetaParitions by partition keys or a subset of those keys
+        if dispatch_by:
+            merged_partitions = base_df.groupby(dispatch_by)
+        else:
+            merged_partitions = base_df.groupby(dataset_factory.partition_keys)
         merged_partitions = merged_partitions["__partition__"].unique()
         for row, labels in merged_partitions.iteritems():
```

Comment:
This is where you iterate over the groupby. Iterating (potentially w/out the iteritems) yields a tuple (group_name, group)
where group_name is the unique value of the groupby statement (e.g. B=1) and the group is the DF

Reply:
You need to extract the logical predicate from row here, I believe